code/modules/antagonists/revenant/revenant_abilities.dm

```diff
@@ -276,52 +278,59 @@
 	name = "Report this to a coder"
 	var/reveal = 80 //How long it reveals the revenant in deciseconds
 	var/stun = 20 //How long it stuns the revenant in deciseconds
-	var/locked = TRUE //If it's locked and needs to be unlocked before use
+	var/unlocked = FALSE //revenant needs to pay essence to learn their ability
 	var/unlock_amount = 100 //How much essence it costs to unlock
 	var/cast_amount = 50 //How much essence it costs to use
 
 /obj/effect/proc_holder/spell/aoe_turf/revenant/Initialize(mapload)
 	. = ..()
-	if(locked)
-		name = "[initial(name)] ([unlock_amount]SE)"
+	update_button_info()
+
+/obj/effect/proc_holder/spell/aoe_turf/revenant/proc/update_button_info()
+	if(unlocked)
+		action.name = "[initial(name)][cast_amount ? " ([cast_amount]E to cast)" : ""]"
 	else
-		name = "[initial(name)] ([cast_amount]E)"
+		action.name = "[initial(name)][unlock_amount ? " ([unlock_amount]SE to learn)" : ""]"
+	action.UpdateButtonIcon()
 
 /obj/effect/proc_holder/spell/aoe_turf/revenant/can_cast(mob/living/simple_animal/revenant/user = usr)
 	if(charge_counter < charge_max)
 		return FALSE
-	if(!istype(user)) //Badmins, no. Badmins, don't do it.
+	if(!isrevenant(user)) // If you're not a revenant, it works anyway.
```

Comment:
Revenant spells being used by non-revenants needs testing evidence that they work properly. 

These were initially disabled because non-revenants do not have the necessary variables for the abilities to work properly as far as I know. Ensure the effects and cooldowns all work properly for humanoid and simplemobs both.

Reply:
The original coder intended it already and I just made the code cleaner

Reply:
also, It mistook to post the image but I checked it's working as well.

Reply:
I'm willing to trust that you tested them